### PR TITLE
Reduce scope

### LINKED ISSUE #69: Reduce scope
# General Support Request

The scope of auro-menu as it currently stands is addressing too wide a scope. 

## Support request

Reduce the current scope of auro-menu to support the primary use case of a menu with one-level indentation support with a checked option. 

Screen Shot 2022-01-20 at 9 27 38 AM

The option to support an unchecked version will be explored in a later iteration.

### LINKED ISSUE #72: Selecting a menu item functionality is split between `mousedown` and `mouseup`
## Describe the bug

Here is a [surge](https://dropdownmenu-demo1.surge.sh/) of menu inside auro-dropdownmenu.

The only thing odd was that the checkmark doesn't show until `mouseup` but the rest of the selected styles show on `mousedown`. It feels a little off when thinking about the experience of it. I think we should be applying both on mousedown.

Playing with this more I released there is actual a further bug tied to this. Using the surge demo open a dropdownmenu and then:

## To Reproduce

1. Click (and hold down the mouse) on any menu item
1. Move the mouse to a different menu item
1. Release the mouse button

Result: The highlight/select colors are applied to the menuoption from step 1 but we never actually complete the rest of the steps tied to making a selection. This results in a menu that looks like this screenshot (you can do this with just menu, no need for dropdownmenu but the side affects are a bit more apparent in dropdownmenu).

## Expected behavior

The steps above should not leave the menu in an odd sort of half selected state. I would expect that all select behavior fires on either `mouseup` or `mousedown`.

## Screenshots

Screen Shot 2022-01-20 at 4 22 11 PM

### PR DESCRIPTION
**Pull request by blackfalcon**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #69 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

The scope of this commit is to remove code that is no longer in scope of this release. Being an individual commit, we can go back and return code as needed as we continue to support these features moving forward.

## Type of change:

Please delete options that are not relevant.

- [ ] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

**Comment by jason-capsule42** [comment]
The basic example imported in the readme.md needs to be updated to match the current API.

**Comment by jason-capsule42** [comment]
Something I find odd with the current implementation is that the select color styles are applied on `mousedown` but the checkmark doesn't show up until `mouseup`. We should aim to have both occur on `mousedown` I think.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]
Readme API example doesn't match current functionality.

**Comment by blackfalcon** [comment]
> Something I find odd with the current implementation is that the select color styles are applied on `mousedown` but the checkmark doesn't show up until `mouseup`. We should aim to have both occur on `mousedown` I think.

The functionality that you are pointing out, as it turns out is expected behavior. 

> An element receives a click event when a pointing device button (such as a mouse's primary mouse button) is both **pressed and released** while the pointer is located inside the element.

While I agree that this is a slightly weird experience, an update supports both the `click` and a `mousedown` event.

**Comment by blackfalcon** [comment]
@jason-capsule42 see last two commits, these address the issues you raised.

File: docs/api.md [hunk 1]
```diff
@@ -40,13 +22,13 @@ The auro-menu element provides users a way to select one option from a pre-defin
...
 ## Properties
 
-| Property     | Attribute    | Type      | Description                              |
-|--------------|--------------|-----------|------------------------------------------|
-| `hideBottom` | `hideBottom` | `Boolean` | Removes hr line from bottom of sub-menu. |
-| `hideTop`    | `hideTop`    | `Boolean` | Removes hr line from top of sub-menu     |
+| Property   | Attribute  | Type      | Description                                      |
+|------------|------------|-----------|--------------------------------------------------|
+| `disabled` | `disabled` | `Boolean` | When true specifies that the menu-option is disabled. |
+| `selected` | `selected` | `boolean` |                                                  |
```
Comment (jason-capsule42) [inline docs/api.md:34]: This should be private shouldn't it?

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]
Small note about the `selected` property, I think that should be private?

Otherwise this looks great.

File: docs/api.md (hunk 1 above)
Comment (blackfalcon) [reply docs/api.md:34]: This is being addressed in https://github.com/AlaskaAirlines/auro-menu/issues/48

**Review by jason-capsule42 — APPROVED** [review]

**Pull request merged**